apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet/Trace/CloudTrace.cs

```diff
@@ -119,16 +117,10 @@ public static void Initialize(string projectId, HttpApplication application, Tra
         private void BeginRequest(object sender, EventArgs e)
         {
             TraceHeaderContext headerContext = TraceHeaderContextUtils.CreateContext(HttpContext.Current.Request);
-            TraceOptions headerOptions = _headerFactory.CreateOptions(headerContext);
-
             // If the trace header says to trace or if the rate limiter allows tracing continue.
-            if (!headerOptions.ShouldTrace)
+            if (!headerContext.ShouldTrace ?? false || !_rateFactory.CreateOptions().ShouldTrace)
```

Comment:
Instead of `!headerContext.ShouldTrace ?? false`, use `headerContext.ShouldTrace != true` or `!headerContext.ShouldTrace.GetValueOrDefault()` (same comment below).